api.php

```diff
@@ -387,46 +387,44 @@ function resolveTags($tagscached)
 	return $tags;
 }
 
-function listOutOfDateMods($modsWithVersions)
-{
-	$outOfDateMods = array();
-	$modids = array_keys($modsWithVersions);
-	$latestReleases = getLatestReleases($modids);
-	foreach($latestReleases as $latestRelease) {
-		$modid = $latestRelease["modidstr"];
-		$version = $modsWithVersions[$modid];
-		if (!empty($latestRelease) && $latestRelease["modversion"] > $version) {
-			$outOfDateMods[$modid] = $latestRelease;
-		}
-	}
-	good(array("statuscode" => 200, "updates" => $outOfDateMods));
-}
-
-function getLatestReleases($modids) {
+function listOutOfDateMods($modsWithVersions) {
 	global $con;
 
-	$latestReleaseInfo = array();
+	$combine = function($modid, $version) {
+		return "$modid@$version";
+	};
+
+	$modids = array_keys($modsWithVersions);
+	$versions = array_values($modsWithVersions);
+	$combined = array_map($combine, $modids, $versions);
+	$combinedParams = implode(",", array_fill(0, count($combined), "?"));
+	$modidParams = implode(",", array_fill(0, count($modids), "?"));
+
+	// Since this query's join on `latestrelease` limits the output to only the latest release for each mod,
+	// We can filter out releases where the incoming version param is the same as the latest release
+	// This means we can use the query itself to fetch only mods which are not on the latest version :)
+	$outOfDateMods = array();
 	$latestReleases = $con->getAll("
 		select
-			`release`.*,
-			asset.*
+			`release`.releaseid,
+			`release`.modidstr,
+			`release`.modversion,
+			`release`.created,
+			`release`.assetid,
+			asset.tagscached
 		from 
 			`release` 
 			join asset on (asset.assetid = `release`.assetid)
 			join (select max(created) as created, modid from `release` group by modid) latestrelease on (latestrelease.created = `release`.created)

```

Comment:
oh, on second glance, this is not fully correct. Upload date does not necessarily correlated with it being the latest version.